.github/workflows/ci.yaml

```diff
@@ -69,6 +74,22 @@ jobs:
           # as the placeholder for the auth token
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
+      - name: Use snapshot version
+        ## This step only runs for "snapshot" branches starting with "snapshot/"
+        ## We have to snapshot-version again in this job because the version
+        ## change in build-frontend only impacts the compiled assets. We need
+        ## the correct version again here for the publish to work properly.
+        if: ${{ github.event_name == 'push' && startsWith(github.ref_name, 'snapshot/') }}
+        run: ./scripts/ui_release.sh --snapshot-version "${{ github.ref_name }}" "${{ github.sha }}" && cd ./ui && npm install
+      - name: Publish snapshot libraries
+        ## This step only runs for "snapshot" branches starting with "snapshot/"
+        if: ${{ github.event_name == 'push' && startsWith(github.ref_name, 'snapshot/') }}
```

Comment:
Just to be extra-sure because I'm still new-ish to github actions, can a reviewer validate for me that this will only be able to run when a maintainer does this (i.e. we can't have any shenanigans with someone external trying to do something malicious with it).

Reply:
yeah that should work